openc3/spec/utilities/aws_bucket_spec.rb

```diff
@@ -22,40 +23,62 @@
 module OpenC3
   describe AwsBucket do
     before(:all) do |example|
+    # These tests work if there's a local S3 or a MINIO service available. To enable
+    # access to MINIO for testing, change the compose.yaml services stanza to:
+    #
+    # services:
+    #   openc3-minio:
+    #     ports:
+    #       - "127.0.0.1:9000:9000"
+      begin
+        sock = Socket.new(Socket::Constants::AF_INET, Socket::Constants::SOCK_STREAM, 0)
+        sock.bind(Socket.pack_sockaddr_in(9000, '127.0.0.1')) #raise if listening
+        sock.close
+        local_s3()
+        Logger.info("No S3 listener - using local_s3 client")
+      rescue Errno::EADDRINUSE;
+        Logger.info("Found listener on port 9000; presumably Minio")
+      end
+
       @bucket = Bucket.getClient.create("bucket#{rand(1000)}")
-    # These tests only work if there's an actual MINIO service available to talk to
-    # Thus we'll just skip them all if we get a networking error
-    # To enable access to MINIO for testing change the compose.yaml file and add
-    # the following to services: openc3-minio:
-    #   ports:
-    #     - "127.0.0.1:9000:9000"
     rescue Seahorse::Client::NetworkingError, Aws::Errors::NoSuchEndpointError => e
+    # We'll just skip them all if we get a networking error.
       example.skip e.message
     end
 
     after(:all) do
       Bucket.getClient.delete(@bucket) if @bucket
+      local_s3_unset()
     end
 
     let(:client) { Bucket.getClient() }
 
-    describe "create, exist?, delete" do
-      it "creates, checks, and deletes a bucket" do
+    describe "basically it" do
```

Comment:
Revert or change to better describe statement

Reply:
Reverted